nolearn/lasagne/base.py

```diff
@@ -359,11 +365,15 @@ def _check_good_input(self, X, y=None, labels=None):
         if self.regression and y is not None and y.ndim == 1:
             y = y.reshape(-1, 1)
 
-        if labels is not None:
-            if len(labels) != x_len:
-                raise ValueError("labels are not the same size as X.")
+        if info is not None:
+            if len(info) != x_len:
+                raise ValueError("X and info are not of equal length.")
+            
+        if label_key is not None:
+            if not info or [True for info_dict in info if label_key not in info_dict]:
```

Comment:
I think you mean any([label_key not in..., but I take your point.

On Tue, May 24, 2016 at 3:31 PM, Scott Mayer McKinney <
notifications@github.com> wrote:

> In nolearn/lasagne/base.py
> <https://github.com/enlitic/nolearn/pull/4#discussion_r64486731>:
>
> > @@ -359,11 +365,15 @@ def _check_good_input(self, X, y=None, labels=None):
> >          if self.regression and y is not None and y.ndim == 1:
> >              y = y.reshape(-1, 1)
> >
> > -        if labels is not None:
> > -            if len(labels) != x_len:
> > -                raise ValueError("labels are not the same size as X.")
> > +        if info is not None:
> > +            if len(info) != x_len:
> > +                raise ValueError("X and info are not of equal length.")
> > +
> > +        if label_key is not None:
> > +            if not info or [True for info_dict in info if label_key not in info_dict]:
>
> if not info or not all([label_key in info_dict for info_dict in info])
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/enlitic/nolearn/pull/4/files/f669694b78133861767c141b2166eac17a71a131#r64486731>
>